scrapy_spider_metadata/_metadata.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any, Dict, Type
+
+from scrapy import Spider
+
+from scrapy_spider_metadata._params import Args
+
+ATTR_NAME = "metadata"
+
+
+def get_metadata_for_spider(spider_cls: Type[Spider]) -> Dict[str, Any]:
```

Comment:
No API is provided for passing the spider name, because that would require passing either a SpiderLoader instance or a CrawlerRunner instance.

Reply:
I am not sure I understand what you mean here, isn’t `getattr(spider_cls, "name", None)` the spider name?

Now that you mention it, I wonder if we should inject the spider name into the metadata. Not sure, though.

Reply:
I meant a function that takes the spider name and uses SpiderLoader to find the class.